### ISSUE TITLE
Long Delay and Errors

### ISSUE THREAD

**Issue by miqrogroove** [humhub-contrib/gallery#75]
When I look at the gallery grid in my profile and click on one of them to open the photo grid, there is a delay of several seconds where it looks like nothing is happening.  If I click and then just wait about 10 seconds, sometimes it will open and sometimes it will do nothing at all.  Occasionally it will say "An internal server error occurred."  Some of my galleries are now unavailable as a result.

I'm using Chrome for Windows, HumHub 1.7.1, and Gallery 1.2.2.

This looks relevant:

```
[13-Jan-2021 08:08:25 America/Detroit] PHP Fatal error:  Maximum execution time of 30 seconds exceeded in /var/www/humhub/protected/vendor/imagine/imagine/lib/Imagine/Imagick/Image.php on line 242
```

**Comment by buddh4** [comment]
@luke- do you think this may be related with the new image compression?

**Comment by luke-** [comment]
What image manipulation method do you use? (GD, ImageMagick, Gmagick)

**Comment by miqrogroove** [comment]
Is that listed in the admin panel somewhere?

**Comment by luke-** [comment]
![image](https://user-images.githubusercontent.com/4736168/106151014-318b8c00-617c-11eb-8e56-aae73cd22fc9.png)

**Comment by miqrogroove** [comment]
Yes, sorry I was looking for something more specific.  Under Prerequisites it says I have both GD and ImageMagick installed.  The error I mentioned at the top is in the Imagick directory, so I'm just guessing it's not GD.

**Comment by buddh4** [comment]
The error seems to be related to Imagick: `/var/www/humhub/protected/vendor/imagine/imagine/lib/Imagine/Imagick/Image.php on line 242`

**Comment by luke-** [comment]
Ok right, so everything looks good there. Imagemagick is the best choice.

Probably then the server resources (CPU, RAM) are not sufficient to resize images fast enough.

**Comment by miqrogroove** [comment]
i3-7100U with 8 GB.  It's not expensive, but it's a dedicated server.  And I'm not getting any memory errors from PHP.

**Comment by luke-** [comment]
Hmm thats strange, can you try to increase the max. execution time?

**Comment by miqrogroove** [comment]
I could do that, but I'm never going to wait 30 seconds for a page load.  Is this script processing thumbnails and caching them?  Is there a way to make it happen asynchronously instead of waiting for the entire gallery to regenerate in Imagick?

**Comment by miqrogroove** [comment]
Also, is there a simple way to switch to GD to see if it runs faster?

**Comment by luke-** [comment]
Normally thumbnails are only created new images and only once. 
However, there has been a change which requires new thumbnail files.

There is currently no console option.

You need to disable the Imagick Extension. But I think this is the fastes one.

**Comment by miqrogroove** [comment]
Going to check if any galleries won't load after repeat attempts.  Some of the smaller ~40 photo galleries still show some placeholder thumbnails on the first attempt but then load fine on the second try.  No new errors yet.

**Comment by miqrogroove** [comment]
Looks like everything is back to normal after clicking around more.  I wonder if this kind of gallery regeneration would be better experienced as a cron task?  Something that could be batched during idle time and work itself out over a day or two instead of just being semi-broken until enough people click on it.
